app/src/main/java/com/example/sharewhatyoucanproject/rdashboard/rdetailedPosts/PostDetailsFragment.kt

```diff
@@ -37,14 +49,65 @@ class PostDetailsFragment : Fragment() {
     ): View? {
         _binding = FragmentPostDetailsBinding.inflate(inflater, container, false)
         val post = Json.decodeFromString<PostModel>(args.postModel)
+        circularProgressIndicator = binding.progressCircular
+        requestbtn = binding.requestbtn
 
         binding.rtitle.text = post.title
         binding.rdesc.text = post.desc
         binding.rdetailimg.load(post.image)
 
+        postDetailsViewModel.name = post.name
```

Comment:
Could you just set the post here instead of all the parameters individually?

Reply:
done